pkg/providers/hcl/intermediate.go

```diff
@@ -149,35 +153,34 @@ type Resource struct {
 	Error        *ParameterMap `hcl:"error,block"`
 }
 
-// OnError intermediate specification
+// OnError intermediate specification.
 type OnError struct {
 	Schema string        `hcl:"schema,optional"`
 	Params *BlockOptions `hcl:"params,block"`
 	Body   hcl.Body      `hcl:",remain"`
 }
 
-// Function intermediate specification
+// Function intermediate specification.
 type Function struct {
-	Name       string        `hcl:"name,label"`
-	Input      *ParameterMap `hcl:"input,block"`
-	References []Resources   `hcl:"resources,block"`
-	Resources  []Resource    `hcl:"resource,block"`
-	Output     *ParameterMap `hcl:"output,block"`
+	Resources `hcl:",remain"`
+
+	Name   string        `hcl:"name,label"`
+	Input  *ParameterMap `hcl:"input,block"`
+	Output *ParameterMap `hcl:"output,block"`
 }
```

Comment:
It seems that this interface is not used anywhere. I think it is best if we remove it for now.

Reply:
removed